snuba/api.py

```diff
@@ -66,6 +68,32 @@ def check_clickhouse():
 )
 
 
+@application.errorhandler(BadRequest)
+def handle_bad_request(exception: BadRequest):
+    cause = getattr(exception, '__cause__', None)
+    if cause is not None and isinstance(cause, json.errors.JSONDecodeError):
+        data = {'error': {'type': 'json', 'message': str(cause)}}
+    elif cause is not None and isinstance(cause, jsonschema.ValidationError):
+        data = {'error': {
+            'type': 'schema',
+            'message': cause.message,
+            'path': list(cause.path),
+            'schema': cause.schema,
+        }}
```

Comment:
This changes the error message a little bit (for the better, in my opinion.) It only tells you the subschema that failed to validate, and the path to the attribute, rather than the full schema on any validation error:

```
{
    "error": {
        "message": "None is not one of ['outcomes', 'events', 'groupedmessage']",
        "path": [
            "dataset"
        ],
        "schema": {
            "enum": [
                "outcomes",
                "events",
                "groupedmessage"
            ]
        },
        "type": "schema"
    }
}
```